File: 탑.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int[] solution(int[] heights)
        {
            int[] answer = new int[heights.Length];
            Stack<int> stack = new Stack<int>(heights);
            int popNum = stack.Pop();
            for(int i=stack.Count-1;i>=0;i--)
            {
                if (popNum < heights[i])
                {
                    answer[stack.Count] = i+1;
                    popNum = stack.Pop();
                    i = stack.Count;
                    continue;
                }
                else if (i == 0)
                {
                    answer[stack.Count] = 0;
                    popNum = stack.Pop();
                    i = stack.Count; 
                    continue;
                }
            }
            return answer;
        }
}

File: 탑.py
def solution(heights):
    answer = []
    for i in range(len(heights)-1,-1,-1):
        for j in range(i-1,-1,-1):
            if heights[i] < heights[j]:
                answer.append(j+1)
                break
            elif j==0:
                answer.append(0)
                break
    answer.append(0)
    return list(reversed(answer))